web/src/components/Korrel8rPanel.tsx

```diff
@@ -34,32 +45,52 @@ export default function Korrel8rPanel() {
   const { korrel8rQueryFromURL } = useURLState();
   const [query, setQuery] = React.useState(korrel8rQueryFromURL);
   const [result, setResult] = React.useState<Result | null>(null);
+  const [showQuery, setShowQuery] = React.useState(false);
+  const [queryType, setSearchType] = React.useState(QueryType.Neighbour);
+  const [depth, setDepth] = React.useState(3);
+  const [goal, setGoal] = React.useState('');
 
   React.useEffect(() => {
     // Set result = null to trigger a reload, don't run the query till then.
     if (result !== null) {
       return;
     }
-    const { request, abort } = getNeighborsGraph({ query });
+    let fetch: CancellableFetch<Korrel8rGraphResponse>;
+    if (queryType === QueryType.Neighbour) {
```

Comment:
I'm not sure if Korrel8r has any more API's that we will want to add in the future, but if there are more then we might want to move this logic over into a new function in the korrel8r-client file instead.

Reply:
None planned. Agreed this would need refactor if it gets any more complicated.